From 092ce0fbd755a56fe2fd25a28519fe974ebf1d2f Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 23 Aug 2013 18:46:06 +0200
Subject: [PATCH] cgroup.c: check return value of unit_realize_cgroup_now()

do not recurse further, if unit_realize_cgroup_now() failed
---
 src/core/cgroup.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/core/cgroup.c b/src/core/cgroup.c
index 5a1c3ad..50b17f3 100644
--- a/src/core/cgroup.c
+++ b/src/core/cgroup.c
@@ -432,8 +432,13 @@ static int unit_realize_cgroup_now(Unit *u) {
                 return 0;
 
         /* First, realize parents */
-        if (UNIT_ISSET(u->slice))
-                unit_realize_cgroup_now(UNIT_DEREF(u->slice));
+        if (UNIT_ISSET(u->slice)) {
+                int r;
+
+                r = unit_realize_cgroup_now(UNIT_DEREF(u->slice));
+                if (r < 0)
+                        return r;
+        }
 
         /* And then do the real work */
         return unit_create_cgroups(u, mask);
